pkg/controller/windowsmachine/windowsmachine_controller.go

```diff
@@ -181,6 +180,27 @@ func (r *ReconcileWindowsMachine) Reconcile(request reconcile.Request) (reconcil
 		return reconcile.Result{}, nil
 	}
 
+	// Get the private key that will be used to configure the instance
+	privateKey, err := secrets.GetPrivateKey(kubeTypes.NamespacedName{Namespace: r.opNamespace,
+		Name: secrets.PrivateKeySecret}, r.client)
+	if err != nil {
+		if k8sapierrors.IsNotFound(err) {
+			// Private key was removed, requeue
+			return reconcile.Result{}, errors.Wrapf(err, "%s does not exist, please create it", secrets.PrivateKeySecret)
+		}
+		return reconcile.Result{}, errors.Wrapf(err, "unable to get secret %s", request.NamespacedName)
+	}
+	// Update the signer with the existing privateKey
+	newSigner, err := signer.Create(privateKey)
```

Comment:
`newSigner` is a misleading as it could be a function name. How about `signer` or `privateKeySigner`? Or just use `r.signer` directly.

Reply:
+1 for `privateKeySigner` or `desiredKeySigner`